tokens/src/lib.rs

```diff
@@ -509,6 +512,23 @@ pub mod module {
 				Ok(())
 			})
 		}
+
+		#[pallet::weight(0)]
+		pub fn sweep_dust(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			accounts: Vec<T::AccountId>,
+		) -> DispatchResult {
+			T::SweepOrigin::ensure_origin(origin)?;
+			for account in accounts {
+				let balance = Self::free_balance(currency_id, &account);
```

Comment:
should be total balances

Reply:
@xlc this will end up calling `do_transfer` which transfers free balance so if account has frozen or reserved it will remain in storage. Maybe we need to make sure `total_balance` is equal to `free_balance` before calling `on_dust` otherwise account will still exist

Reply:
ok. can just require reserved balance is zero.